### DISCUSSION TITLE
Curvatures for each axis

### DISCUSSION THREAD

**Discussion by jkissi** [daavoo/pyntcloud#313] [Q&A] [answered]
Hi @daavoo 
Another question I am hoping you can help me with. Your library produces a mean curvature measurement, which is great. I was wondering if there is a way to get the curvature for each axis x, y, z as well. If not, do you know how I could derive it using the results of the functions in your library? 

Once again, thanks for your help here 
Cheers 
j

**Comment by daavoo** [comment, marked as answer]
Hi @jkissi I'm not sure if I fully understand the concept of `curvature for each axis`, could you please elaborate? Maybe you just want the `eigenvectors` (directions) and `eigenvalues` (magnitudes)? (per https://en.wikipedia.org/wiki/Principal_curvature) ?

What pyntcloud currently computes as `curvature` is a formula based on `eigenvalues` (check https://github.com/daavoo/pyntcloud/blob/master/examples/%5Bscalar_fields%5D%20eigenvalues.ipynb for the formula).

**Reply by jkissi** [reply]
Hi @daavoo 
I am working on the assumption that your curvature method is is equivalent to the average change in direction in x, y, z of each point relative to its neighbours. What I am also after is value for the change in x, y and z separately for each point. 

Do you know of a way to breakdown the overall curvature, so as to get the mean curve in the x direction for each for, for example? I hope that what I am saying makes sense...

Thanks for the link also btw

**Reply by jkissi** [reply]
I can take it from your delayed answer that you have nothing more to add. Anyway thanks for you help so far, very much appreciated! 
j
